src/basic/time-util.c

```diff
@@ -1605,19 +1605,27 @@ bool clock_supported(clockid_t clock) {
 }
 
 int get_timezone(char **ret) {
-        _cleanup_free_ char *t = NULL;
+        _cleanup_free_ char *t = NULL, *link_dir = NULL, *abspath = NULL;
         int r;
 
         assert(ret);
 
-        r = readlink_malloc("/etc/localtime", &t);
+        r = readlink_malloc(etc_localtime(), &t);
         if (r == -ENOENT)
                 /* If the symlink does not exist, assume "UTC", like glibc does */
                 return strdup_to(ret, "UTC");
         if (r < 0)
                 return r; /* Return EINVAL if not a symlink */
 
-        const char *e = PATH_STARTSWITH_SET(t, "/usr/share/zoneinfo/", "../usr/share/zoneinfo/");
+        r = path_extract_directory(etc_localtime(), &link_dir);
+        if (r < 0)
+                return r;
+
+        abspath = path_make_absolute(t, link_dir);
+        if (!abspath)
+                return -ENOMEM;
+
+        const char *e = path_startswith(abspath, "/usr/share/zoneinfo/");
```

Comment:
I do not think this works as expected.
The file is used not only by systemd, and must be a direct symbolic link to a zone file.

Reply:
Only if you need to get the name of the zone.

Reply:
For applications that need to read the symlink, and indirect link to from `/etc/localtime` to the writable symlink works. There are cases of course where applications try to read the path to extract the name of the timezone instead of querying `org.freedesktop.timedate1`. But those are corner case, usually they have their own timezone db, and they have a hack to match the zoneinfo name to their db.

We have been using this since 2020 on Ubuntu Core, and we have not yet encountered an issue. (I am not saying this design is great).

I am not sure I understand why using `CHASE_EXTRACT_FILENAME` here. The zone name might have a directory components, like "Etc/UTC". Here we would get only "UTC". Unless I misunderstood.

Reply:
If CHASE_EXTRACT_FILENAME is not suitable here, then please forget the flag.